### ISSUE TITLE
[Question] Webhooks [[LIVENESS]] and [[FAILURE]]

### ISSUE THREAD

**Issue by CarljohanDanling** [Xabaril/AspNetCore.Diagnostics.HealthChecks#415]
Hi. If I have a collection of healthchecks under one specific HealthCheckEndpoint. Is it possible in the [[LIVENESS]] and [[FAILURE]] to target the specific one that Webhooks notifies Slack about?

For example. If I have 3 apps under the endpoint name "Agents" (app1, app2 & app3). If app1 fails, the [[LIVENESS]] says "Agents are failing...". I would like to point out that just app1 is failing, not the collection of Agents.

Any ideas how to do this?

**Comment by CarlosLanderas** [comment]
Hello @CarljohanDanling. What do you think about being able to customize if the payload should be notified, the message and the descriptions using funcs where you receive the UIReport, something like this:

```csharp
setup
.AddWebhookNotification("webhook1",
uri: "status/200?code=ax3rt56s", payload: "{ message: \"Webhook report for [[LIVENESS]]: [[FAILURE]]\"}",
restorePayload: "{ message: \"[[LIVENESS]] is back to life\"}",
shouldNotifyFunc: report => DateTime.UtcNow.Hour > 8 && DateTime.UtcNow.Hour < 20
customMessageFunc: (report) =>
{
    var failing = report.Entries.Where(e => e.Value.Status == UIHealthStatus.Unhealthy);
    return $"{failing.Count()} healthchecks are failing";
}, 
customDescriptionFunc: report =>
{
    var failing = report.Entries.Where(e => e.Value.Status == UIHealthStatus.Unhealthy);
    return $"HealthChecks with names {string.Join("/", failing.Select(f => f.Key))} are failing";
});

```

**Comment by CarljohanDanling** [comment]
That looks good!

**Comment by CarlosLanderas** [comment]
I've already coded that implementation last night. So if everyone finds it interesting we will add this new customization feature

**Comment by CarlosLanderas** [comment]
This feature will be available in UI version 3.0.11
